data/Store.js

```diff
@@ -947,6 +965,25 @@ function forIn(obj, fn) {
     }
 }
 
+/**
+ * Modes for how record field values should be considered when filtering.
```

Comment:
Maybe worth an extra emphasis on FieldFilter here:

```suggestion
 * Determines which Record field values should be used for matching against a `FieldFilter`.
```